tools/serverpod_cli/lib/src/analyzer/dart/endpoint_analyzers/endpoint_parameter_analyzer.dart

```diff
@@ -43,54 +44,69 @@ abstract class EndpointParameterAnalyzer {
   static List<SourceSpanSeverityException> validate(
     List<ParameterElement> parameters,
   ) {
-    return parameters
-        .map((parameter) {
-          var type = parameter.type;
-          if (type.isDartAsyncFuture) {
-            return SourceSpanSeverityException(
-              'The type "Future" is not a supported endpoint parameter type.',
-              parameter.span,
-            );
-          }
-
-          if (type.isDartAsyncStream && type is ParameterizedType) {
-            if (type.nullabilitySuffix != NullabilitySuffix.none) {
-              return SourceSpanSeverityException(
-                'Nullable parameters of the type "Stream" are not supported.',
-                parameter.span,
-              );
-            }
-
-            var typeArguments = type.typeArguments;
-            if (typeArguments.length != 1) {
-              // Streams only allow a single generic so this case is only here for safety.
-              return SourceSpanSeverityException(
-                'The type "Stream" must have exactly one type argument. E.g. Stream<String>.',
-                parameter.span,
-              );
-            }
-            var innerType = typeArguments[0];
-            if (innerType is VoidType) {
-              return SourceSpanSeverityException(
-                'The type "Stream" does not support void generic type.',
-                parameter.span,
-              );
-            }
-          }
-
-          try {
-            TypeDefinition.fromDartType(parameter.type);
-          } on FromDartTypeClassNameException catch (e) {
-            return SourceSpanSeverityException(
-              'The type "${e.type}" is not a supported endpoint parameter type.',
-              parameter.span,
-            );
-          }
-
-          return null;
-        })
-        .whereType<SourceSpanSeverityException>()
-        .toList();
+    List<SourceSpanSeverityException> exceptions = [];
+
+    bool firstParameterIsSession = parameters.isNotEmpty &&
+        parameters.first.type.element?.displayName == Keyword.sessionClassName;
+
+    if (firstParameterIsSession) {
+      bool sessionIsNullable =
+          parameters.first.type.nullabilitySuffix != NullabilitySuffix.none;
+      if (sessionIsNullable) {
+        exceptions.add(SourceSpanSeverityException(
+            'The "Session" argument in an endpoint method does not have to be nullable, consider removing the "?".',
```

Comment:
Suggestion: Change to your suggested proposal of `consider making it non-nullable`

Reply:
[dce77e1](https://github.com/serverpod/serverpod/pull/2597/commits/dce77e12dce032b4967912fada3137685b0d90fe)

Reply:
![Uploading image.png…]()

Reply:
![Uploading image.png…]()